### PR TITLE
[ONNX] Unsqueeze operator

### PR DESCRIPTION
**Pull request by tsocha**

### CODE REVIEW

File: src/ngraph/frontend/onnx_import/op/unsqueeze.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                std::iota(std::begin(input_order), std::end(input_order), 0);
+
+                for (auto axis : axes)
+                {
+                    if (axis < 0 || axis > data_shape.size())
+                    {
+                        throw error::parameter::Value(
+                            "Unsqueeze", node.get_name(), "provided axes attribute is not valid.");
+                    }
+                    data_shape.insert(data_shape.begin() + axis, 1);
```
Comment (apwojcik) [inline src/ngraph/frontend/onnx_import/op/unsqueeze.cpp]: use `std::next(std::begin(data_shape), axis)` instead.

File: src/ngraph/frontend/onnx_import/op/unsqueeze.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+                // Generate an increasing sequence (0,1,2,3..) as input_order for Reshape
+                ngraph::AxisVector input_order(data_shape.size());
+                std::iota(std::begin(input_order), std::end(input_order), 0);
+
+                for (auto axis : axes)
+                {
+                    if (axis < 0 || axis > data_shape.size())
+                    {
+                        throw error::parameter::Value(
```
Comment (apwojcik) [inline src/ngraph/frontend/onnx_import/op/unsqueeze.cpp]: Can we skip invalid attribute (just reporting warning) and continue with other attributes (robustness principle), or invalid attribute means we can not continue as it is now?

File: src/ngraph/frontend/onnx_import/op/unsqueeze.hpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <memory>
```
Comment (apwojcik) [inline src/ngraph/frontend/onnx_import/op/unsqueeze.hpp]: `memory` is not needed here. Only header files which define types required for the interface. Here we have only `ngraph::NodeVector` and `onnx_import::Node`.

File: test/onnx_import.cpp [hunk 4]
```diff
@@ -522,6 +522,29 @@ TEST(onnx, model_sub)
     EXPECT_TRUE(test::all_close_f(expected_output, result_vectors.front()));
 }
 
+TEST(onnx, model_unsqueeze)
+{
+    auto function = ngraph::onnx_import::import_onnx_function(
```
Comment (apwojcik) [inline test/onnx_import.cpp]: remove `ngraph::`

File: test/onnx_import.cpp [hunk 5]
```diff
@@ -522,6 +522,29 @@ TEST(onnx, model_sub)
...
+        ngraph::file_util::path_join(SERIALIZED_ZOO, "onnx/unsqueeze.onnx"));
+
+    Inputs inputs;
+    inputs.emplace_back(ngraph::test::NDArray<float, 3>(
```
Comment (apwojcik) [inline test/onnx_import.cpp]: remove `ngraph::`

File: test/onnx_import.cpp [hunk 6]
```diff
@@ -522,6 +522,29 @@ TEST(onnx, model_sub)
...
+                            {{{1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}},
+                             {{1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}},
+                             {{1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}}})
+                            .get_vector());
+
+    auto expected_output =
+        ngraph::test::NDArray<float, 4>(
```
Comment (apwojcik) [inline test/onnx_import.cpp]: remove `ngraph::`

File: test/onnx_import.cpp [hunk 7]
```diff
@@ -522,6 +522,29 @@ TEST(onnx, model_sub)
...
+        ngraph::file_util::path_join(SERIALIZED_ZOO, "onnx/unsqueeze.onnx"));
+
+    Inputs inputs;
+    inputs.emplace_back(ngraph::test::NDArray<float, 3>(
+                            {{{1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}},
+                             {{1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}},
+                             {{1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}}})
+                            .get_vector());
+
+    auto expected_output =
```
Comment (apwojcik) [inline test/onnx_import.cpp]: Use `Output expected_outputs{....}`. We want unit test source code to be uniform.

File: test/onnx_import.cpp [hunk 8]
```diff
@@ -522,6 +522,29 @@ TEST(onnx, model_sub)
...
+        ngraph::test::NDArray<float, 4>(
+            {{{{1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}},
+              {{1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}},
+              {{1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}, {1, 1, 1, 1, 1}}}})
+            .get_vector();
+
+    auto result_vectors = execute(function, inputs, "INTERPRETER");
```
Comment (apwojcik) [inline test/onnx_import.cpp]: Use `Output outputs{...}` instead `auto result_vectors`.

File: src/ngraph/frontend/onnx_import/op/unsqueeze.hpp [hunk 9]
```diff
@@ -0,0 +1,36 @@
+//*****************************************************************************
```
Comment (apwojcik) [inline src/ngraph/frontend/onnx_import/op/unsqueeze.hpp]: clang format

File: src/ngraph/frontend/onnx_import/op/unsqueeze.cpp [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <numeric>
+
+#include "unsqueeze.hpp"
```
[the remark this answers is not in the document]
Comment (tsocha) [reply src/ngraph/frontend/onnx_import/op/unsqueeze.cpp]: #includes are separated by empty line on topics.

File: src/ngraph/frontend/onnx_import/op/unsqueeze.hpp (hunk 9 above)
Comment (tsocha) [reply src/ngraph/frontend/onnx_import/op/unsqueeze.hpp]: Clang-format-3-9 passed

**Pull request merged**